extensions/ql-vscode/src/extension.ts

```diff
@@ -112,7 +117,7 @@ export async function activate(ctx: ExtensionContext): Promise<void> {
         await noUpdatesLoggingFunc("CodeQL CLI is installed externally so could not be updated.");
         break;
       case DistributionUpdateCheckResultKind.UpdateAvailable:
-        if (beganMainExtensionActivation) {
+        if (beganMainExtensionActivation || !allowAutoUpdating) {
           const updateAvailableMessage = `Version "${result.updatedRelease.name}" of the CodeQL CLI is now available. ` +
             "The update will be installed after Visual Studio Code restarts. Restart now to upgrade?";
```

Comment:
This messaging is no longer correct:

```suggestion
            "Do you wish to upgrade?";
```